### DISCUSSION TITLE
24 hours time editing and view

### DISCUSSION THREAD

**Discussion by skobkin** [firefly-iii/firefly-iii#9371] [General]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [x] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/explanation/more-information/what-its-not/)
- [x] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

- Add option to use 24 hour mode

This would be very useful for regions with such locale standards

### Solution

- Add option to use 24 hour date/time picker mode, it should enable 24 hour mode in DateTime picker and 24 hours format in views

### Additional context

https://en.wikipedia.org/wiki/24-hour_clock

**Comment by JC5** [comment]
Hi, thanks for opening an issue!

This is not on me, but a browser setting. Please set your browser to something not USA based.

**Comment by skobkin** [comment]
How can I set my browser to English without selecting 12-hour clock format?
OS (Windows, Linux) allows to tweak locale settings, but AFAIK, browser doesn't allow to do that.

I think multiple web services allow to choose clock format in settings for this reason.

**Comment by JC5** [comment]
I'm happy to hear that. I use standard html elements. 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

**Comment by skobkin** [comment]
@JC5 I see.

I've tried to double check how it works in my browser (Vivaldi). I've found that it has some language settings.
It inherits system locale settings by default:
![Vivaldi language settings](https://github.com/user-attachments/assets/3a3726b7-9b45-4261-8898-a904d3dc6742)

But it still doesn't affect Firefly III:
![Firefly III date selection 12-hours](https://github.com/user-attachments/assets/554759ed-531f-448a-a693-f2715da01f8c)
